### PR TITLE
recursive mkdir

### PR DESCRIPTION
**Pull request by SG5**
Tested only at `BlockingDriver` and `EioDriver` because i can't compile `php-uv`

### CODE REVIEW

File: lib/Driver.php [hunk 1]
```diff
@@ -129,9 +129,10 @@ public function unlink($path);
      *
      * @param string $path
      * @param int $mode
+     * @param bool $recursive
      * @return \Amp\Promise
      */
-    public function mkdir($path, $mode = 0644);
+    public function mkdir($path, $mode = 0644, $recursive);
```
Comment (kelunik) [inline lib/Driver.php]: This should default to `false`, just like the native API: http://php.net/mkdir

File: lib/EioDriver.php [hunk 2]
```diff
@@ -356,11 +356,39 @@ private function onUnlink($data, $result, $req) {
     /**
      * {@inheritdoc}
      */
-    public function mkdir($path, $mode = 0644) {
+    public function mkdir($path, $mode = 0644, $recursive) {
```
Comment (kelunik) [inline lib/EioDriver.php]: Same here about the default value.

File: lib/EioDriver.php [hunk 3]
```diff
@@ -356,11 +356,39 @@ private function onUnlink($data, $result, $req) {
...
         \call_user_func($this->incrementor, 1);
         $promisor = new Deferred;
         $priority = \EIO_PRI_DEFAULT;
-        \eio_mkdir($path, $mode, $priority, [$this, "onGenericResult"], $promisor);
+
+        if ($recursive) {
+            $arrayPath = array_filter(explode(DIRECTORY_SEPARATOR, $path));
```
Comment (kelunik) [inline lib/EioDriver.php]: I guess we should first replace all `/` with `DIRECTORY_SEPARATOR`, otherwise `/` won't work on Windows.

**Comment by kelunik** [comment]
Hey @SG5, thanks for the PR! I think we just need minor changes until we can merge it.

**Comment by SG5** [comment]
All change requests were pushed

**Comment by kelunik** [comment]
@bwoebi LGTM, please merge if it looks good to you as well.

**Comment by SG5** [comment]
I just noticed what Hacktoberfest will be started 1st October, can you hold this merge for next week?

**Comment by kelunik** [comment]
Sure, I just guess that opening it will count, not it being merged.

**Pull request merged**